test/language/module-code/dynamic-import/catch/top-level-import-catch-eval-script-code-target.js

```diff
@@ -0,0 +1,38 @@
+// This file was procedurally generated from the following sources:
+// - src/dynamic-import/eval-script-code-target.case
+// - src/dynamic-import/catch/top-level.template
+/*---
+description: import() from a module code can load a file with script code, but the target is resolved into a Module Record (top level)
+esid: sec-import-call-runtime-semantics-evaluation
+features: [dynamic-import]
+flags: [generated, module, async]
+info: |
+    ImportCall :
+        import( AssignmentExpression )
+
+    1. Let referencingScriptOrModule be ! GetActiveScriptOrModule().
+    2. Assert: referencingScriptOrModule is a Script Record or Module Record (i.e. is not null).
+    3. Let argRef be the result of evaluating AssignmentExpression.
+    4. Let specifier be ? GetValue(argRef).
+    5. Let promiseCapability be ! NewPromiseCapability(%Promise%).
+    6. Let specifierString be ToString(specifier).
+    7. IfAbruptRejectPromise(specifierString, promiseCapability).
+    8. Perform ! HostImportModuleDynamically(referencingScriptOrModule, specifierString, promiseCapability).
+    9. Return promiseCapability.[[Promise]].
+
+
+    Modules
+
+    Static Semantics: Early Errors
+
+      ModuleBody : ModuleItemList
+      - It is a Syntax Error if the LexicallyDeclaredNames of ModuleItemList containsany duplicate entries.
+      - It is a Syntax Error if any element of the LexicallyDeclaredNames of ModuleItemList also occurs in the VarDeclaredNames of ModuleItemList.
+
+---*/
+
+import('./script-code_FIXTURE.js').catch(error => {
+
```

Comment:
How does an implementation or interpretation detect that fixtures should be treated as script or module goals?

Reply:
The `script-code_FIXTURE` currently only evaluates _(without error in the script goal)_. Test files gave metadata to indicate `module` or script (lack of `module` specified)` goals, but I don't see FIXTURE scripts having such indicators.

Reply:
Ahhh, on further testing I get what you're saying. Sorry for the mixup.

Reply:
So script code is treated as module?
Is the limitation for `await` keyword, html  comments, and strict mode in place too?

Reply:
that's a weird thing. The script code is not exactly treated as module but it needs to create a Module Record. You can't have duplicate lexical bindings in a Module Record and that's where we get the error.

I'll check out a verification with the html comments.